cpp/include/tensorrt_llm/common/cudaUtils.h

```diff
@@ -311,6 +311,16 @@ inline int getSMVersion()
     return sm;
 }
 
+inline int getSMFamily()
+{
+    int sm = getSMVersion();
+    if (sm == 100 || sm == 103)
+    {
+        return 100;
+    }
+    return sm;
```

Comment:
A Nit. 
Why are we only overriding the 103? not others, like 80-86, like 100 and 120?
Shall we rename it to a more spefic function like `isSM100Family`?